bot/exts/filtering/_filter_lists/antispam.py

```diff
@@ -0,0 +1,191 @@
+            new_infraction.infraction_reason = (
+                f"{triggers[0].name.replace('_', ' ')} spam – {ctx.filter_info[triggers[0]]}"
+            )
+            current_actions["infraction_and_notification"] = new_infraction
+            self.message_deletion_queue[ctx.author].current_infraction = new_infraction.infraction_type
+        else:
+            current_actions.pop("infraction_and_notification", None)
+
+        # Provide some message in case another filter list wants there to be an alert.
+        return current_actions, ["Handling spam event..."], {ListType.DENY: triggers}
+
+    def _create_deletion_context_handler(self, context_id: Member) -> Callable[[FilterContext], Coroutine]:
+        async def schedule_processing(ctx: FilterContext) -> None:
+            """
+            Schedule a coroutine to process the deletion context.
+
+            It cannot be awaited directly, as it waits ALERT_DELAY seconds, and actioning a filtering context depends on
+            all actions finishing.
+
+            This is async and takes a context to adhere to the type of ctx.additional_actions.
+            """
+            async def process_deletion_context() -> None:
+                """Processes the Deletion Context queue."""
+                log.trace("Sleeping before processing message deletion queue.")
+                await asyncio.sleep(ALERT_DELAY)
+
+                if context_id not in self.message_deletion_queue:
+                    log.error(f"Started processing deletion queue for context `{context_id}`, but it was not found!")
+                    return
+
+                deletion_context = self.message_deletion_queue.pop(context_id)
+                await deletion_context.send_alert(self)
+
+            scheduling.create_task(process_deletion_context())
+
+        return schedule_processing
+
+
+@dataclass
+class DeletionContext:
+    """Represents a Deletion Context for a single spam event."""
+
+    contexts: list[FilterContext] = field(default_factory=list)
+    rules: set[UniqueFilter] = field(default_factory=set)
+    current_infraction: Infraction | None = None
+
+    def add(self, ctx: FilterContext, rules: list[UniqueFilter]) -> None:
+        """Adds new rule violation events to the deletion context."""
+        self.contexts.append(ctx)
+        self.rules.update(rules)
+
+    async def send_alert(self, antispam_list: AntispamList) -> None:
+        """Post the mod alert."""
+        if not self.contexts or not self.rules:
+            return
+
+        webhook = antispam_list.filtering_cog.webhook
+        if not webhook:
+            return
+
+        ctx, *other_contexts = self.contexts
+        new_ctx = FilterContext(ctx.event, ctx.author, ctx.channel, ctx.content, ctx.message)
+        new_ctx.action_descriptions = reduce(
+            add, (other_ctx.action_descriptions for other_ctx in other_contexts), ctx.action_descriptions
+        )
```

Comment:
When spamming a bunch I noticed I can get a response like this, would it make sense to remove duplicates? (ignore location of comment, not sure where this would be done).

![image](https://user-images.githubusercontent.com/22353562/222919588-3c6fcce0-71c5-4271-b5ab-dc820e286e5b.png)

Reply:
In this case, would something like `deleted all, muted, deleted X 9` be ok?

Reply:
Yeah sounds good

Reply:
[Group identical action descriptions in antispam](https://github.com/python-discord/bot/pull/2390/commits/7a1a553b67ff7fc300e9b457f8750efbf28ecf96)